src/Http/Authentication.Abstractions/src/IAuthenticationConfigurationProvider.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.AspNetCore.Authentication;
+
+/// <summary>
+/// Provides an interface for implmenting a construct that provides
+/// access to specific configuration sections.
+/// </summary>
+public interface IAuthenticationConfigurationProvider
+{
+    /// <summary>
+    /// Gets the root configuration managed by the provider.
+    /// </summary>
+    IConfigurationRoot Configuration { get; }
```

Comment:
Why is this exposed?

Reply:
We need to access parts of the configuration that aren't defined in our JSON-based config (like signing secrets) in the implementations for our scheme registerations.